tests/foreman/ui/test_ldap_authentication.py

```diff
@@ -1257,3 +1257,18 @@ def test_external_new_user_login_and_check_count(session):
         finally:
             update_rhsso_settings_in_satellite(revert=True)
             delete_rhsso_user(user_details['username'])
+
+
+@tier2
+def test_positive_test_connection_functionality(session, ldap_data, ipa_data):
+    """Verify for a positive test connection response
+
+    :id: 5daf3976-9b5c-11ea-96f8-4ceb42ab8dbc
+
+    :steps: Assert test connection of AD and IPA.
+
+    :expectedresults: Positive test connection of AD and IPA
+    """
+    with session:
+        for ldap_host in (ldap_data['ldap_hostname'], ipa_data['ldap_ipa_hostname']):
+            assert not session.ldapauthentication.test_connection({'ldap_server.host': ldap_host})
```

Comment:
I still don't like the way this is checked. That assert implies the function can return some meaningful value. But that's not the case, it *always* returns None. Therefore, there are only two possibilities: 1) exception thrown, or 2) exception not thrown. I think the assert is meaningless here and you can just run the test_connection function and either get an exception or not.

Reply:
Yes. That make sense.
I have made the changes now.

Reply:
@akhil-jha add a block of code which verifies asserts there was no error thrown. e.g. 

```
with pytest.raises(Exception):
    assert not session.ldapauthentication.test_connection({'ldap_server.host': ldap_host})
```
@lhellebr I agreewith you, let me know if you agree on this ? please give some example

Reply:
@omkarkhatavkar , your suggestion would check that the function *does* raise exception, am I right? That's not what should happen, the exception should *not* be raised. So you can either 1) Leave it as it is after the latest update (i.e. just call the function) or catch the exception if it happens and do something about it (perhaps re-raise it). I think both is fine.